WebService/v1/Controllers/DeviceModelScriptsController.cs

```diff
@@ -0,0 +1,169 @@
+using Microsoft.Azure.IoTSolutions.DeviceSimulation.WebService.v1.Filters;
+using Microsoft.Azure.IoTSolutions.DeviceSimulation.WebService.v1.Models;
+using Microsoft.Azure.IoTSolutions.DeviceSimulation.WebService.v1.Models.Helpers;
+using Microsoft.Azure.IoTSolutions.DeviceSimulation.WebService.v1.Models.DeviceModelScriptApiModel;
+
+namespace Microsoft.Azure.IoTSolutions.DeviceSimulation.WebService.v1.Controllers
+{
+    [ExceptionsFilter]
+    public class DeviceModelScriptsController : Controller
+    {
+        private readonly string ApplicationJavascript = "application/javascript";
+        private readonly ILogger log;
+        private readonly IDeviceModelScripts simulationScriptService;
+
+        public DeviceModelScriptsController(
+            IDeviceModelScripts simulationScriptService,
+            ILogger logger)
+        {
+            this.simulationScriptService = simulationScriptService;
+            this.log = logger;
+        }
+
+        [HttpGet(Version.PATH + "/[controller]")]
+        public async Task<DeviceModelScriptListModel> GetAsync()
+        {
+            return DeviceModelScriptListModel.FromServiceModel(await this.simulationScriptService.GetListAsync());
+        }
+
+        [HttpGet(Version.PATH + "/[controller]/{id}")]
+        public async Task<DeviceModelScriptApiModel> GetAsync(string id)
+        {
+            return DeviceModelScriptApiModel.FromServiceModel(await this.simulationScriptService.GetAsync(id));
+        }
+
+        [HttpPost(Version.PATH + "/[controller]!validate")]
+        public async Task<ActionResult> Validate(IFormFile file)
+        {
+            if (file == null)
+            {
+                this.log.Warn("No data provided");
+                throw new BadRequestException("No data provided.");
+            }
+
+            if (file.ContentType != this.ApplicationJavascript)
+            {
+                this.log.Warn("Wrong content type provided", () => new { file.ContentType });
+                throw new BadRequestException("Wrong content type provided.");
+            }
+
+            try
+            {
+                var parser = new JavaScriptParser();
+                var reader = new StreamReader(file.OpenReadStream());
+                var rawScript = reader.ReadToEnd();
+                parser.Parse(rawScript);
+            }
+            catch (Exception e)
+            {
+                var result = new JsonResult(new ValidationApiModel
+                    {
+                        IsValid = false,
+                        Messages = new List<string>
+                        {
+                            e.Message
+                        }
+                    })
+                    { StatusCode = (int) HttpStatusCode.BadRequest };
+                return await Task.FromResult(result);
+            }
+
+            return await Task.FromResult(new JsonResult(new ValidationApiModel()) { StatusCode = (int) HttpStatusCode.OK });
+        }
```

Comment:
the syntax of lines 82 and 85 seems overly complex, any way to simplify?

Reply:
Done.